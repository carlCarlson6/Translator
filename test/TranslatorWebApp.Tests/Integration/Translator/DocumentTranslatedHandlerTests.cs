using Contracts.Events;
using FluentAssertions;
using Rebus.Handlers;
using Snapshooter.Xunit;
using TranslatorWebApp.Common.Core;
using TranslatorWebApp.Common.Core.Errors;
using TranslatorWebApp.Common.Infrastructure.AzureStorageTables;
using TranslatorWebApp.Tests.TestHelpers;
using TranslatorWebApp.TranslatorWorker.TextTranslation;
using Xunit;

namespace TranslatorWebApp.Tests.Integration.Translator;

public class DocumentTranslatedHandlerTests : TestWithAzurite
{
    [Fact]
    public async Task WhenHandleDocumentTranslated_ThenDocStatusIsUpdatedToCompleted()
    {
        var tableClient = await InitTableClientAsync();
        var doc = new TranslationDocument(
            FakeGuidGenerator.New(),
            new TranslationText("this text should be translated into spanish"),
            new TranslationText("esta es la traduccion"),
            Status.InProcess);
        await tableClient.AddEntityAsync(TranslationDocumentTableEntity.From(doc));

        await BuildHandler().Handle(new DocumentTranslated(doc.Id, doc.OriginalText.ToString()));

        var response = await tableClient.GetEntityAsync<TranslationDocumentTableEntity>("Translation", doc.Id.ToString());
        response.Value.Status.Should().Be(Status.Completed);
        response.Value.Should().MatchSnapshot(o => o.IgnoreField(nameof(TranslationDocumentTableEntity.Timestamp)));
    }

    [Fact]
    public async Task GivenNoTranslationDocument_WhenHandleDocumentTranslated_ThenTranslationDocumentNotFoundIsThrown()
    {
        await InitTableClientAsync();
        var act = () => BuildHandler().Handle(new DocumentTranslated(FakeGuidGenerator.New(), "some text"));
        await act.Should().ThrowAsync<TranslationDocumentNotFound>();
    }

    private IHandleMessages<DocumentTranslated> BuildHandler() => new DocumentTranslatedHandler(
        new AzureStorageTableTranslationDocumentsRepository(GivenTableClient()));
}