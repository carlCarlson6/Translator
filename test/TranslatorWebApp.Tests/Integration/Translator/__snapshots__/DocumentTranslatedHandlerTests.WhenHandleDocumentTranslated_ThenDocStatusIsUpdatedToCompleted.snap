{
  "PartitionKey": "Translation",
  "RowKey": "a4395f86-cab1-406f-acd1-274c27894b93",
  "Status": "Completed",
  "OriginalText": "this text should be translated into spanish",
  "TranslatedText": "this text should be translated into spanish",
  "Timestamp": "2022-10-08T23:55:28.942+00:00",
  "ETag": {}
}
